docs/topics/spiders.rst

```diff
@@ -136,6 +136,13 @@ scrapy.Spider
        attributes in the new instance so they can be accessed later inside the
        spider's code.
 
+       .. versionchanged:: VERSION
+
+           The settings in ``crawler.settings`` can now be modified, which is
+           handy if you want to modify them based on arguments. As a
+           consequence, these settings aren't the final values as they can be
+           modified later by e.g. :ref:`add-ons <topics-addons>`.
```

Comment:
Something like this maybe? I'm not sure about "when the spider starts" point, is there a more precise definition?

```suggestion
           The settings in ``crawler.settings`` can be modified in the 
           ``from_crawler`` method, which is handy if you want to modify 
           them based on spider arguments. As a consequence, the settings 
           aren't the final values as they can be modified later by 
           e.g. :ref:`add-ons <topics-addons>`. The settings
           become final when the spider starts.
```

Reply:
Rather than discussing when the settings become final, we could change the angle and instead tell users how they can get the earliest possible access to the final settings from a spider (start_requests? some signal?).

Reply:
I think `ExecutionEngine()` doesn't fire any signals so that would be `start_requests()`.